tests/test-buffer.js

```diff
@@ -9,14 +9,28 @@ var buff;
 var lens = [[-1, 0],
             [0, 0],
             [8, 8],
-            [8.5, 8],
-            [4294967295, 4294967295],
-            [4294967296, 4294967295]];
-for(var i = 0; i < lens.length; i++) {
+            [8.5, 8]];
+for (var i = 0; i < lens.length; i++) {
     buff = new Buffer(lens[i][0]);
     assert(buff.length === lens[i][1],
-           "The length of Buffer(" + lens[i][0] + ") expected:" + lens[i][1] +
-           " got:" + buff.length);
+           "Creating buffer of length " + lens[i][0] + " expected:" +
+           lens[i][1] + " got:" + buff.length);
+}
+
+// thrown errors are allowed on these lengths
+lens = [[4294967295, 4294967295],
+        [4294967296, 4294967295]];
```

Comment:
I think we'd better to use 1073741823(2^30-1) following Node.js 6.x [buffer.kMaxLength
](https://nodejs.org/dist/latest-v6.x/docs/api/buffer.html#buffer_buffer_kmaxlength), and it's expected to be a valid length for large memory(>1G) or throw error when devices have no enough available memory.

1073741824(2^30) should always throw error for out of memory.